### ISSUE TITLE
Owned side of OneToOne relation not defined in generated query

### ISSUE THREAD

**Issue by royibernthal** [typeorm/typeorm#1111]
I'm trying to SELECT rows in request WHERE ticket equals some id number. ticket is an entity.

I have 2 entities with a OneToOne relation - ticket and request.

ticket/request is the owning side of the relation - it has JoinColumn.
On the other hand, request/ticket is the owned side of the relation - it doesn't have JoinColumn.

When I try to add a where statement of request in the ticket entity (the owning side of the relation), there's no problem.

However when I try to do the opposite, I'm getting the error:
>  Unknown column 'request.ticket' in 'where clause'

After taking a look at the generated query, it seems all the entity's parameters are defined in the SELECT, except for ticket which is not defined in the select and only added to the where clause. (since I'm calling queryBuilder.where()).

> SELECT request.id AS request_id, request.type AS request_type, request.date AS request_date, request.session_type AS request_session_type, request.session AS request_session, request.game AS request_game FROM request request WHERE request.session_type = :session_type AND request.game = :game AND **request.ticket = :ticket**

**In other words, createQueryBuilder doesn't define the owned side of the OneToOne relation in the generated query.**

It seems like a bug, or am I missing something?

**Comment by pleerock** [comment]
can you please provide a complete code example with reproduction of this issue (but better if you create a failing test if you can)

**Comment by royibernthal** [comment]
I looked into it more, I think it starts to make sense, the problem is really the WHERE clause in which I add the ticket column which doesn't REALLY exist in request table, it only exists in concept since it's in the owned side of the relation.

Is such a thing even supposed to work in TypeORM?

```
@Table()
export class Ticket {

    @PrimaryGeneratedColumn()
    id: number;

    @OneToOne(type => Request, {
        cascadeInsert: true,
        cascadeUpdate: true
    })
    @JoinColumn()
    request: Request;

}

@Table()
export class Request {

    @PrimaryGeneratedColumn()
    id: number;

    @OneToOne(type => Ticket, ticket => ticket.request, {
        cascadeInsert: true,
        cascadeUpdate: true,
        cascadeRemove: true
    })
    ticket: Ticket;

}

this.connection.getRepository(Request).createQueryBuilder('request')
			.where('request.ticket = :ticket')
			.setParameter('ticket', 2)
			.leftJoinAndSelect('request.ticket', 'ticket')
			.getOne();
```

**Comment by pleerock** [comment]
What typeorm version are you using? Looking on `@Table()` decorator usage looks like you are using a very old version. I strongly recommend you to upgrade orm version since lot of things are changed and fixed since those versions.

Change your code this way and it will work:

```ts
this.connection.getRepository(Request).createQueryBuilder('request')
			.leftJoinAndSelect('request.ticket', 'ticket',)
			.where('ticket.id = :ticket')
			.setParameter('ticket', 2)
			.getOne();
```

**Comment by royibernthal** [comment]
True, I was using a very old version. I migrated to the newest version, the changes look good, and your suggestion solved the problem.

I'm now trying to export the db's SQL from my dev env and importing it into my prod env.
Naturally, the SQL was created by TypeORM.
When importing it, I'm getting the following error:

> ERROR 1064 (42000) at line 29: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6),
>   `relationId` int(11) DEFAULT 0
> ) ENGI' at line 4

```
CREATE TABLE `tableName` ( //line 29
  `id` int(11) NOT NULL,
  `name` varchar(255) NOT NULL,
  `date` datetime(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6),
  `relationId` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```

Any idea why the SQL syntax is wrong? Let me know if you'd like me to open a separate issue for this.

**Comment by pleerock** [comment]
Can you provide an entity which generate this code?

**Comment by royibernthal** [comment]
```
@Entity()
export class SomeClass {

    @PrimaryGeneratedColumn()
    id: number;
    
    @Column()
    name: string;
    
    @CreateDateColumn()
    date: Date;

    @ManyToOne(type => RelationClass, relation => relation.someClasses)
    relation: RelationClass;

}
```

**Comment by pleerock** [comment]
What about `RelationClass` model code? I don't see any problems in your code that could lead errors you have. Are you able to create a failing test (send a PR with a test that fails, just see how others do it in pull requests section) or at least a code repository with reproduction of this problem

**Comment by royibernthal** [comment]
Here's a ZIP - npm start to sync the entities. I already added for you there the sql exported by phpMyAdmin.

I have no problems importing it to phpMyAdmin, but I'm getting the error I mentioned above when I try to import it via MySQL Workbench, something that didn't happen before I migrated to the new version.

https://drive.google.com/file/d/1N0XzsLmNXWtqOvCUERQz6NHI7dffaWy_/view?usp=sharing

**Comment by pleerock** [comment]
can you please create a git repo, we are on the github, right?) Also why are you talking about sql? I don't need sql - the only thing I need is a minimal reproduction code example - entities, connection creation code, ormconfig and package.json - that's all. I'll need to do pull, npm install, edit config and run - and I must see a problem they way u see it.

**Comment by royibernthal** [comment]
Git repo:
https://github.com/royibernthal/typeorm_repro

I'm talking about SQL because the problem is not when TypeORM is running, there can't be a failing test or anything of the sort, at least not the one you have in mind.

The error is thrown when I'm importing to MySQL Workbench tables that were created by TypeORM (with synchronize set to true) after I migrated to the new version.

In other words, you could run the code repro but the problem is mainly the SQL it generates in the end. Makes sense?

**Comment by royibernthal** [comment]
I looked into it more, and discussed it with ClearDB support. Let me rephrase the issue:

The problem is that the relevant ClearDB clusters run mysql 5.5, for mysql 5.6.5 and below:

"CURRENT_TIMESTAMP is only acceptable on TIMESTAMP fields. DATETIME fields must be left either with a null default value, or no default value at all - default values must be a constant value, not the result of an expression."

With that said, synchronize does throw an error so I have to disable it when in production and modify the tables manually.

Does TypeORM has anything to do with the mysql version used or is this backward compatibility issue entirely related to the mysql module TypeORM is using?

Is there a way to have TypeORM not set the default value at all for Date fields? Or at the very least set it to a constant value and not to an expression like CURRENT_TIMESTAMP.

**Comment by pleerock** [comment]
> Is there a way to have TypeORM not set the default value at all for Date fields? 

it does not do it. It does it only for `@CreateDateColumn` and `@UpdateDateColumn` which are specific decorators. If you simply want a date column you don't need to use those decorators and nothing else will generate you a default with timestamp value.

**Comment by royibernthal** [comment]
Of course, I meant when using these decorators.
For instance, I'd like the date to be updated every time the data is changed, without setting a default value due to the mysql version restriction.
For CreateDateColumn I'm not even sure if it makes sense logically as the implementation itself is most likely setting the default value?

**Comment by pleerock** [comment]
For now you have to use regular date column and maybe using listeners update date before updation. I'll think if we can improve this to support older mysql versions as well.

**Comment by royibernthal** [comment]
I understand. Great, such support would be nice. Thanks for the help.
